### PR TITLE
Glibc backtrace

### PR DESCRIPTION
**Pull request by chvink**
This function can help with debugging in some situations. With some extra work I can include a solaris and windows implementation, too.

### CODE REVIEW

File: modules/c++/sys/source/Backtrace.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#include <sys/Backtrace.h>
+
+#include <sys/sys_config.h>
+#include <sstream>
+
+const size_t MAX_STACK_ENTRIES = 62;
```
Comment (asylvest) [inline modules/c++/sys/source/Backtrace.cpp]: Add `static`

File: modules/c++/sys/source/Backtrace.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+#ifdef HAVE_EXECINFO_H
+
+#include <execinfo.h>
+#include <cstdlib>
+
+std::string sys::getBacktrace()
+{
+    void * stackBuffer[MAX_STACK_ENTRIES];
```
Comment (asylvest) [inline modules/c++/sys/source/Backtrace.cpp]: Spacing here and in other spots with pointers should be `void* stackBuffer`

File: modules/c++/sys/source/Backtrace.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    int currentStackSize = backtrace(stackBuffer, MAX_STACK_ENTRIES);
+    char ** stackSymbols = backtrace_symbols(stackBuffer, currentStackSize);
+
+    std::stringstream ss;
+    for (int ii = 0; ii < currentStackSize; ++ii)
+    {
+        ss << stackSymbols[ii] << std::endl;
+    }
+    std::free(stackSymbols);
```
Comment (asylvest) [inline modules/c++/sys/source/Backtrace.cpp]: Use a RAII object so this can't leak

**Comment by chvink** [comment]
The feedback should all be addressed.

**Comment by asylvest** [comment]
So this is really cool.  I didn't know such a function existed.  Changes look good.  @chvink can you add an issue to support this on Windows too?  We don't need to do it now but this way we won't forget.

FYI to @clydestanfield @JDanielSmith @mwysockiMDA @msciaramitaro @jbruce2112 @mbojrab

**Pull request merged**